### ISSUE TITLE
No Documentation of Any Sort

### ISSUE THREAD

**Issue by Symbolics** [r0man/hive-el#3]
Even a few bullet points would be nice.

Installed from a package, added (require 'hive) to .emacs and tried to get an inferior shell. 'Not a known product' error. 

Update to most recent version of hive.el; same problem.

**Comment by r0man** [comment]
You should be able to just run `sql-hive` and get a prompt as
long as `hive` is in your path. Not sure what's going on
here. Which version of Emacs are you using? I can't reproduce
this with Arch Linux's Emacs version and also not with the git
master branch. But I have seen this problem before. I have the
suspicion that something is strange with `sql-add-product`. Could
you try it with this 8178395fa6dbae91e70b62a7cf4bd074a593c485?

**Comment by Symbolics** [comment]
I am using 0.1.1, with the modified requires clause.

I’m running 24.3.1 on MacOS. It does appear that sql-add-product isn’t quite working, because if I run the code in eval-after-load manually I get the same result.
- Steve

**Comment by Symbolics** [comment]
Any suggestions on how to debug this? I’ve tried on a few fresh installations, installing from package-install (MELPA). Could that repository be stale? Would love to have an emacs/hive environment.

**Comment by Symbolics** [comment]
A bit more debugging reveals that (eval-after-load “sql” .. isn’t running. Manually evaluating (sql-add-product …) adds hive to the list of products, but then SQLi reports: SQL product hive is not supported; treated as ANSI.

I guess there must not be too many emacs users using hive at the moment. Any other suggestions that don’t involve Eclipse?
- SteveN

**Comment by r0man** [comment]
Die you (require 'hive) ?

**Comment by Symbolics** [comment]
No, I’ll try that next. I now realize though that this isn’t going to work, except in limited circumstances, because often the thick client hive isn’t able to submit job directly to the cluster.

Is it possible to convince hive-mode to use beeline instead of hive?

Cheers,
- Steve

**Comment by r0man** [comment]
Try require. Otherwise the code is not loaded. I thought this is obvious.
I'm not familiar with beeline, but submitting hive jobs to a cluster works.
That's what I'm doing most of the time.

**Comment by tninja** [comment]
never works. Hi r0man. I use require and manually sql-add-product but it never works.

"hive is not a known product"

have you every try this on any other box? I am using a ubuntu 14.04 LTS. If you fix the problem and tell us it would be great.

**Comment by snunez1** [comment]
Any progress made here? It's been 7 months since first reported and I was hoping that a fix might have been posted...

**Comment by krhys** [comment]
There a bug in sql-add-product.

This line:     (add-to-list 'sql-product-alist `((,product :name ,display . ,plist)))

should be:    (add-to-list 'sql-product-alist `(,product :name ,display . ,plist))

**Comment by r0man** [comment]
Patch welcome!

**Comment by krhys** [comment]
Looks like it's been fixed upstream. Only an issue for peoples stuck on
older versions of emacs.
